### DISCUSSION TITLE
Sub entity support

### DISCUSSION THREAD

**Discussion by heavyweight87** [enbility/eebus-go#103] [Q&A] [answered]
Hi,
As some devices have sub entities  (heatpump  entity with compressor sub entitty for example) I wondered if there was an easy way to support sub entities? From what I see there currently isn't support for this

Do you think there is an easy way to add it?

**Comment by DerAndereAndi** [comment, marked as answer]
Hi, this is already supported.

Example code for an EV added to an EVSE:

```go
localDevice := c.eebusService.LocalDevice()
evseEntity := localDevice.EntityForType(model.EntityTypeTypeEVSE)

// EV
evseAddress := evseEntity.Address().Entity[0]
evEntity := spine.NewEntityLocal(localDevice, model.EntityTypeTypeEV, []model.AddressEntityType{evseAddress, 1})
```

**Reply by heavyweight87** [reply]
Thanks! I think this awesome repo could maybe do with some detailed examples (and docs). Mind if I add some?

**Reply by DerAndereAndi** [reply]
Please feel free to suggest something.
